relation/src/gadgets/cmp.rs

```diff
@@ -200,115 +288,134 @@ mod test {
-        Ok(())
-    }
-    fn test_is_ge<F: PrimeField>(a: &F, b: &F) -> Result<(), CircuitError> {
-        let mut circuit = PlonkCircuit::<F>::new_turbo_plonk();
-        let expected_result = if a > b { F::one() } else { F::zero() };
-        let a = circuit.create_variable(*a)?;
-        let b = circuit.create_variable(*b)?;
-
-        let c = circuit.is_gt(a, b)?;
-        assert!(circuit.witness(c.into())?.eq(&expected_result));
-        assert!(circuit.check_circuit_satisfiability(&[]).is_ok());
-        Ok(())
-    }
-    fn test_is_geq<F: PrimeField>(a: &F, b: &F) -> Result<(), CircuitError> {
-        let mut circuit = PlonkCircuit::<F>::new_turbo_plonk();
-        let expected_result = if a >= b { F::one() } else { F::zero() };
-        let a = circuit.create_variable(*a)?;
-        let b = circuit.create_variable(*b)?;
-
-        let c = circuit.is_geq(a, b)?;
-        assert!(circuit.witness(c.into())?.eq(&expected_result));
-        assert!(circuit.check_circuit_satisfiability(&[]).is_ok());
-        Ok(())
-    }
-    fn test_enforce_le<F: PrimeField>(a: &F, b: &F) -> Result<(), CircuitError> {
-        let mut circuit = PlonkCircuit::<F>::new_turbo_plonk();
-        let expected_result = a < b;
-        let a = circuit.create_variable(*a)?;
-        let b = circuit.create_variable(*b)?;
-        circuit.enforce_lt(a, b)?;
-        if expected_result {
-            assert!(circuit.check_circuit_satisfiability(&[]).is_ok())
+        let expected_result = if a.cmp(b) == ordering
+            || (a.cmp(b) == Ordering::Equal && should_also_check_equality)
+        {
+            F::one()
         } else {
-            assert!(circuit.check_circuit_satisfiability(&[]).is_err());
-        }
-        Ok(())
-    }
-    fn test_enforce_leq<F: PrimeField>(a: &F, b: &F) -> Result<(), CircuitError> {
-        let mut circuit = PlonkCircuit::<F>::new_turbo_plonk();
-        let expected_result = a <= b;
+            F::zero()
+        };
         let a = circuit.create_variable(*a)?;
-        let b = circuit.create_variable(*b)?;
-        circuit.enforce_leq(a, b)?;
-        if expected_result {
-            assert!(circuit.check_circuit_satisfiability(&[]).is_ok())
+        let c: BoolVar = if is_b_constant {
+            match ordering {
+                Ordering::Less => {
+                    if should_also_check_equality {
+                        circuit.is_leq_constant(a, *b)?
+                    } else {
+                        circuit.is_lt_constant(a, *b)?
+                    }
+                },
+                Ordering::Greater => {
+                    if should_also_check_equality {
+                        circuit.is_geq_constant(a, *b)?
+                    } else {
+                        circuit.is_gt_constant(a, *b)?
+                    }
+                },
+                Ordering::Equal => {
+                    let b = circuit.create_constant_variable(*b)?;
```

Comment:
Do we need this test path?

Reply:
No. Added here only for completeness. Test will not enter here.